Modules/Filtering/Convolution/include/itkFFTConvolutionImageFilter.hxx

```diff
@@ -126,47 +147,160 @@ FFTConvolutionImageFilter<TInputImage, TKernelImage, TOutputImage, TInternalPrec
   ProgressAccumulator *      progress,
   float                      progressWeight)
 {
-  // Pad the image
-  InputSizeType   padSize = this->GetPadSize();
-  InputRegionType inputRegion = input->GetLargestPossibleRegion();
-  InputSizeType   inputSize = inputRegion.GetSize();
+  float remainingProgress = 1.0f;
+
+  InputRegionType  inputLargestRegion = input->GetLargestPossibleRegion();
+  InputSizeType    inputLargestSize = inputLargestRegion.GetSize();
+  InputIndexType   inputLargestIndex = inputLargestRegion.GetIndex();
+  InputRegionType  inputRequestedRegion = input->GetRequestedRegion();
+  InputSizeType    inputRequestedSize = inputRequestedRegion.GetSize();
+  InputIndexType   inputRequestedIndex = inputRequestedRegion.GetIndex();
+  OutputRegionType outputRequestedRegion = this->GetOutput()->GetRequestedRegion();
+  OutputSizeType   outputRequestedSize = outputRequestedRegion.GetSize();
+  OutputIndexType  outputRequestedIndex = outputRequestedRegion.GetIndex();
+
+  // Pad the input image such that the requested region, expanded by
+  // twice the kernel radius, lies entirely within the buffered region.
+  // If the requested region is at least the kernel radius away from
+  // all boundaries of the buffered input region then padding is skipped.
 
   using InputPadFilterType = PadImageFilter<InputImageType, InputImageType>;
-  auto inputPadder = InputPadFilterType::New();
-  inputPadder->SetBoundaryCondition(this->GetBoundaryCondition());
+  using PadSizeType = typename InputPadFilterType::SizeType;
 
-  InputSizeType inputLowerBound = this->GetPadLowerBound();
-  inputPadder->SetPadLowerBound(inputLowerBound);
+  PadSizeType    lowerPad, upperPad;
+  bool           needsKernelPadding = false;
+  KernelSizeType kernelRadius = this->GetKernelRadius();
+  for (unsigned int dim = 0; dim < ImageDimension; ++dim)
+  {
+    signed int lower = static_cast<signed int>(kernelRadius[dim]) - (inputRequestedIndex[dim] - inputLargestIndex[dim]);
+    signed int upper =
+      static_cast<signed int>(kernelRadius[dim]) -
+      ((inputLargestIndex[dim] + inputLargestSize[dim]) - (inputRequestedIndex[dim] + inputRequestedSize[dim]));
+    lowerPad[dim] = (lower > 0 ? lower : 0);
+    upperPad[dim] = (upper > 0 ? upper : 0);
+
+    needsKernelPadding = (lower > 0 || upper > 0);
```

Comment:
padding is overwritten here - only the last dimension matters
```suggestion
    needsKernelPadding = needsKernelPadding || lower > 0 || upper > 0;
```
or `needsKernelPadding ||= lower > 0 || upper > 0;` if it compiles.

Reply:
Good catch, I also just found this in testing. Fixed in latest commit.